modules/planning/tasks/traffic_decider/pull_over.cc

```diff
@@ -73,37 +73,61 @@ bool PullOver::IsValidStop(const PointENU& stop_point,
   return true;
 }
 
-bool PullOver::GetPullOverStop(PointENU* stop_point, double* stop_heading) {
-  auto* pull_over_status = GetPlanningStatus()->
-      mutable_planning_state()->mutable_pull_over();
+int PullOver::GetPullOverStop(PointENU* stop_point, double* stop_heading) {
+  auto&  pull_over_status = GetPlanningStatus()->
+      mutable_planning_state()->pull_over();
   // reuse existing stop point
-  if (pull_over_status->has_stop_point() &&
-      pull_over_status->has_stop_heading()) {
-    if (IsValidStop(pull_over_status->stop_point(),
-                    pull_over_status->stop_heading())) {
-      *stop_point = pull_over_status->stop_point();
-      *stop_heading = pull_over_status->stop_heading();
-      return true;
+  if (pull_over_status.has_stop_point() &&
+      pull_over_status.has_stop_heading()) {
+    if (IsValidStop(pull_over_status.stop_point(),
+                    pull_over_status.stop_heading())) {
+      *stop_point = pull_over_status.stop_point();
+      *stop_heading = pull_over_status.stop_heading();
+      return 0;
     }
   }
+
   // calculate new stop point if don't have a pull over stop
   return SearchPullOverStop(stop_point, stop_heading);
 }
 
-bool PullOver::SearchPullOverStop(PointENU* stop_point, double* stop_heading) {
-  if (CheckPullOver()) {
-    return false;
+int PullOver::SearchPullOverStop(PointENU* stop_point, double* stop_heading) {
+  if (!CheckPullOver()) {
+    return -1;
   }
 
-  return false;
+  // TODO(all): temparily set stop point at lane_boarder
+  common::SLPoint stop_point_sl;
+  const double adc_front_edge_s =
+      reference_line_info_->AdcSlBoundary().end_s();
+  const double stop_s = adc_front_edge_s + config_.pull_over().plan_distance();
+  stop_point_sl.set_s(stop_s);
+
+  const auto& reference_line = reference_line_info_->reference_line();
+  double heading = reference_line.GetReferencePoint(stop_s).heading();
+  double lane_left_width = 0.0;
+  double lane_right_width = 0.0;
+  reference_line.GetLaneWidth(stop_s, &lane_left_width, &lane_right_width);
+  stop_point_sl.set_l(-lane_right_width);
+
+  common::math::Vec2d point;
+  reference_line.SLToXY(stop_point_sl, &point);
```

Comment:
What happen if stop_point_sl.s() > reference_line.length()?

Reply:
stop_point is adc's position + plan_distance(config, currently it is set to 20m). It shall be within reference line.